docs/make-endpoint-toc

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/awk -f
```

Comment:
This file started as an awk-oneliner I used to generate a table of contents based on the API endpoints in the documentation.  I decided, since I thought it was useful, to clean it up and commit it.  I don't expect our API to change *too* much, but I figure this tool maybe useful in the future to automatically parse this endpoint format and create a TOC.

`docs/` may not be the best place for this - I really don't have any emotional investment in this tool either, so i'm open to really anything involving this.